### ISSUE TITLE
[Bug] virtio-mmio kernel cmdline append breaks init arguments separated by `--` 

### ISSUE THREAD

**Issue by leonjza** [firecracker-microvm/firecracker#2709]
# Describe the bug

Currently, it is not possible to pass arguments to init using `--` ([ref](https://www.kernel.org/doc/html/latest/admin-guide/kernel-parameters.html)) as extra arguments are appended after the user provided command line [here](https://github.com/firecracker-microvm/firecracker/blob/cf46cb18dbc0a5dad6d19a69e44a80e21d3e43e3/src/vmm/src/device_manager/mmio.rs#L228-L233). When a user uses `--`, the behaviour causes essential arguments to be ignored as they will be passed to init.

In a bit of a contrived example, but for illustrations sake, let's imagine my init was to `/bin/cat` a file.
When specifying kernel `boot_args` such as `console=ttyS0 nomodules random.trust_cpu=on reboot=k panic=1 pci=off ip=:::::eth0:dhcp init=/bin/cat -- /etc/password`, the  aforementioned code block will append extra arguments which technically get discarded because of the the `--`. The resultant command line would therefore be:

```text
[    0.000000] Command line: console=ttyS0 nomodules random.trust_cpu=on reboot=k panic=1 pci=off ip=:::::eth0:dhcp init=/bin/cat -- /etc/password root=/dev/vda rw virtio_mmio.device=4K@0xd0000000:5 virtio_mmio.device=4K@0xd0001000:6 virtio_mmio.device=4K@0xd0002000:7
```

## To Reproduce

1. Start firecracker with `firecracker --api-sock /tmp/firecracker.socket`.
2. Configure the kernel with the following JSON body:

```json
{
   "kernel_image_path":"/path/to/kern",
   "boot_args":"console=ttyS0 nomodules random.trust_cpu=on reboot=k panic=1 pci=off ip=:::::eth0:dhcp init=/bin/cat -- /etc/password"
}
```
3. Configure other parts to get a VM to boot.
4. Start the instance and observe the `Command line:` output.

## Expected behaviour

Kernel arguments such as `root=` and `virtio_mmio.device=` should appear before user supplied arguments.

## Environment

1. Arch : x86_64
2. OS: Ubuntu 21.04
3. Firecracker 0.24.6

## Checks

- [x] Have you searched the Firecracker Issues database for similar problems?
- [x] Have you read the existing relevant Firecracker documentation?
- [x] Are you certain the bug being reported is a Firecracker issue?

**Comment by dianpopa** [comment]
Hi @leonjza ,

I posted a fix for this issue #2716. Take a look and let me know if this fixes your problem.

**Comment by georgepisaltu** [comment]
Pending on [this PR](https://github.com/rust-vmm/linux-loader/pull/94) in rust-vmm/linux-loader, we will have a fix for this.
